packages/odyssey-react/src/components/Field/Field.tsx

```diff
@@ -0,0 +1,177 @@
+
+interface Props extends SharedFieldTypes {
+  /**
+   * Input to be rendered within the Field
+   */
+  children: ReactElement | ReactElement[];
+
+  /**
+   * The underlying input element id attribute. Automatically generated if not provided
+   */
+  inputId: string;
+
+  /**
+   * The underlying parent semantic HTML element.
+   * @default div
+   */
+  as?: "div" | "fieldset";
+}
+
+interface PropsLabel {
+  inputId: string;
+  optionalLabel?: string;
+  required: boolean;
+  labelHidden?: boolean;
+  children: ReactNode;
+  as?: "label" | "legend";
+}
+interface PropsHint {
+  id: string;
+  children: ReactText;
+}
+interface PropsError {
+  id: string;
+  children: ReactNode;
+}
+
+interface Statics {
+  Label: typeof Label;
+  Hint: typeof Hint;
+  Error: typeof Error;
+}
+
+let Field: FunctionComponent<Props> & Statics = Object.assign(
+  (props: Props) => {
+    const {
+      error,
+      hint,
+      inputId,
+      label,
+      optionalLabel,
+      required = true,
+      children,
+      labelHidden,
+      as = "div",
+    } = props;
+
+    const Tag = as;
+    const TagLabel = as === "fieldset" ? "legend" : "label";
+
+    return (
+      <Tag className={styles.root}>
+        <Field.Label
+          inputId={inputId}
+          required={required}
+          optionalLabel={optionalLabel}
+          labelHidden={labelHidden}
+          as={TagLabel}
+        >
+          {label}
+        </Field.Label>
+        {hint && <Field.Hint id={inputId}>{hint}</Field.Hint>}
+        {children}
+        {error && <Field.Error id={inputId}>{error}</Field.Error>}
+      </Tag>
+    );
+  },
+  {
+    Label,
+    Error,
+    Hint,
+  }
+);
+
+function Label(props: PropsLabel) {
+  const {
+    inputId,
+    optionalLabel = "Optional",
+    required,
+    children,
+    labelHidden,
+    as = "label",
+  } = props;
+
+  const Tag = as;
+
+  const label = (
+    <Tag className={styles.label} htmlFor={inputId}>
+      <Text color="heading" weight="bold">
+        {children}
+      </Text>
+      {!required && optionalLabel && (
+        <span className={styles.optionalLabel}>
+          <Text color="sub" size="caption" weight="regular" lineHeight="normal">
+            optionalLabel
+          </Text>
+        </span>
+      )}
+    </Tag>
+  );
+
+  const labelVisuallyHidden = (
+    <ScreenReaderText>
+      <Tag htmlFor={inputId}>
+        {children}
+        {!required && optionalLabel && <span children={optionalLabel} />}
+      </Tag>
+    </ScreenReaderText>
+  );
+
+  return labelHidden ? labelVisuallyHidden : label;
+}
+
+function Hint({ id, children }: PropsHint) {
+  return (
+    <p className={styles.hint} id={`${id}-hint`}>
+      <Text color="body" size="caption">
+        {children}
+      </Text>
+    </p>
+  );
+}
+
+function Error({ id, children }: PropsError) {
+  return (
+    <p className={styles.error} id={`${id}-error`}>
+      <Text color="danger" size="caption">
+        {children}
```

Comment:
I notice that children is typed as `ReactNode` here (really anything) and `Hint` above has its children typed as `ReactText` - should we unify here? Keep in mind that might be a breaking change...

Reply:
`Error` requires a `ScreenReaderText` be present, so `ReactText` is insufficient.